EIPS/eip-5453.md

```diff
@@ -0,0 +1,131 @@
+MAGIC_WORLD(8bytes)
+```
+
+The total length denoted as `LENGTH_OF_ENDORSEMENT = 137 bytes`
+
+5. The `(r,s,v)` are an ECDSA signature of all method fields plus the head remainder of `extraData` removing the last `LENGTH_OF_ENDORSEMENT` bytes. See the [Reference Implementation](#reference-implementation) for clarification.
+
+## Rationale
+
+1. Choosing [EIP-5269](./eip-5269.md) instead of [EIP-165](./eip-165.md) because the data format cannot be represented by method signature as in [EIP-165](./eip-165.md).
+
+1. Originally we considered adding the endorsement at the first part of `extraData`. We turn into adding the endorsement to the ending of data, which allows the following future extensions such as:
+
+- Allow general endorsable be applied
+- Allow nested endorsements
+
+## Backwards Compatibility
+
+The design assumes a `bytes calldata extraData` to maximize the flexibility of future extensions. This assumption is compatible with [EIP-721](eip-721.md), [EIP-1155](eip-1155.md) and many other ERC-track EIPs. Those that aren't, such as [EIP-20](./eip-20.md), can also be updated to support it, such as using a wrapper contract or proxy upgrade.
+
+## Reference Implementation
+
+```solidity
+// Macro is not supported yet in
+// Implementer can just replicate the following expansion by hand until
+// The macro is supported in solidity.
+#define REMAIN_LENGTH extraData.length - LENGTH_OF_ENDORSEMENT;
+
+abstract contract ERC5453 {
+    modifier onlyEndorsed(bytes32 _msgDigest, bytes calldata _end/**orsement**/) {
+        require(_end.length == LENGTH_OF_ENDORSEMENT);
+        require(_end[_end.length - 8:] == MAGIC_WORLD);
+
+
+        // ERC-1271 prefix. See https://eips.ethereum.org/EIPS/eip-1271
+        string memory erc1271Prefix = "\x19Ethereum Signed Message:\n32";
+        bytes32 erc1271Hash = keccak256(abi.encodePacked(erc1271Prefix, _msgDigest));
+        address endorser = ecrecover(erc1271Hash,
+            uint8(_end[128]), // v
+            bytes32(_end[64:96]), // r
+            bytes32(_eSig[96:128]) // s
+        );
+        // _isEligibleEndorser is application specific.
+        require (_isEligibleEndorser(endorser), "Endorser is not eligible to transfer.");
+        _;
+    }
+}
+
+contract EndorsableERC721 is SomeERC721, ERC5453 {
+  function safeTransferFrom(
+    address from,
+    address to,
+    uint256 id,
+    bytes calldata extraData)
+
+    onlyEndorsed(  // used as modifier
+        keccak256(
+            abi.encodePacked(
+                from, to, id, amount,
+                extraData[:REMAIN_LENGTH], // first part of extraData is reserved for original use for extraData unendorsed.
+                extraData[REMAIN_LENGTH: REMAIN_LENGTH + 32], // nonce of endorsement for the {contract, endorser} combination
+                extraData[REMAIN_LENGTH + 32: REMAIN_LENGTH + 64], // valid_by for the endorsement
+            ),
+        data[REMAIN_LENGTH:] // the endorsement component
```

Comment:
Shouldn't it be `extraData[REMAIN_LENGTH:]`?

Reply:
It might. There is issue in this rrefimpl. Lets talk later today soon